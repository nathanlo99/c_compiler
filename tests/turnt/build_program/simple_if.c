
int wain(int a, int b) {
  if (b % 2 == 0) {
    a = a * a;
  } else {
    a = a + a;
  }
  println(a);
  return a;
}
