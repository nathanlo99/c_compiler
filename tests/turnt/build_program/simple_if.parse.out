procedures main
main INT WAIN LPAREN dcl COMMA dcl RPAREN LBRACE dcls statements RETURN expr SEMI RBRACE
INT int
WAIN wain
LPAREN (
dcl type ID
type INT
INT int
ID a
COMMA ,
dcl type ID
type INT
INT int
ID b
RPAREN )
LBRACE {
dcls .EMPTY
statements statements statement
statements statements statement
statements .EMPTY
statement IF LPAREN test RPAREN LBRACE statements RBRACE ELSE LBRACE statements RBRACE
IF if
LPAREN (
test sum EQ sum
sum term
term term PCT factor
term factor
factor ID
ID b
PCT %
factor NUM
NUM 2
EQ ==
sum term
term factor
factor NUM
NUM 0
RPAREN )
LBRACE {
statements statements statement
statements .EMPTY
statement expr SEMI
expr lvalue BECOMES expr
lvalue ID
ID a
BECOMES =
expr test
test sum
sum term
term term STAR factor
term factor
factor ID
ID a
STAR *
factor ID
ID a
SEMI ;
RBRACE }
ELSE else
LBRACE {
statements statements statement
statements .EMPTY
statement expr SEMI
expr lvalue BECOMES expr
lvalue ID
ID a
BECOMES =
expr test
test sum
sum sum PLUS term
sum term
term factor
factor ID
ID a
PLUS +
term factor
factor ID
ID a
SEMI ;
RBRACE }
statement PRINTLN LPAREN expr RPAREN SEMI
PRINTLN println
LPAREN (
expr test
test sum
sum term
term factor
factor ID
ID a
RPAREN )
SEMI ;
RETURN return
expr test
test sum
sum term
term factor
factor ID
ID a
SEMI ;
RBRACE }
